front/app/containers/MainHeader/DesktopNavbar/getNavbarItemPropsArray.tsx

```diff
@@ -16,8 +16,8 @@ export default function getNavbarItemPropsArray(
   }, {});
 
   return navbarItems.map((navbarItem) => {
-    const linkTo: string = navbarItem.relationships.page
-      ? getPageSlug(pagesById, navbarItem.relationships.page.data.id)
+    const linkTo: string = navbarItem.relationships.static_page.data
+      ? getPageSlug(pagesById, navbarItem.relationships.static_page.data.id)
       : DEFAULT_PAGE_SLUGS[navbarItem.attributes.code];
```

Comment:
If `navbarItem.attributes.code === 'custom'`, `navbarItem.relationships.static_page.data` will always be defined. If `navbarItem.attributes.code !== 'custom'`, `navbarItem.relationships.static_page.data` will always be undefined. I have literally spent hours trying to get TypeScript to understand this, without success. If you know a better way to write this piece of code that is more type-safe, I am happy to change this bit, because I also don't like it

Reply:
There are two things we can do to improve it:
1) if you use a function with if/else blocks it's easier to infer the correct type (it doesn't work with object selectors for some reason, don't know why without further investigating 🤔).
2) make that check explicit for code = 'custom'. So `navbarItem.attributes.code === 'custom' && navbarItem.relationships.static_page.data ? getPageSlug... : ... ; `